pkg/autoscaler/autoscaler.go

```diff
@@ -120,6 +158,19 @@ func (agg *totalAggregation) observedPods(now time.Time) float64 {
 	return podCount
 }
 
+// The number of available pods that are estimated via sample stats.
+func (agg *totalAggregation) estimatedPods(now time.Time) float64 {
```

Comment:
I don't think you're going to want to use estimation for the number of pods.  You will never get the right number for short periods of time.

Instead, you should get the number of READY pods from the Deployment.

Reply:
It is using the number of IPs from the Endpoints, which is the same with the number of READY pods. Sorry for the confusing `estimated*` names. I didn't use the current `observed*` names because the data are samples as part of the all truth. Maybe call them `average*`?

Reply:
> It is using the number of IPs from the Endpoints

Oh, got it.  I thought you were using the number of unique pods you observe.  I think we should make a clean cut to using IPs as a basis for how many pods are "in service".  If you make that change first, it might make this change simpler.